### PR TITLE
Remove the hardcoded directory paths C:\ReactOS

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

JIRA issue: [CORE-14747](https://jira.reactos.org/browse/CORE-14747)

### CODE REVIEW

File: base/services/telnetd/telnetd.c [hunk 1]
```diff
@@ -367,12 +367,19 @@ static void RunShell(client_t *client)
    STARTUPINFO           si;
    PROCESS_INFORMATION   piProcInfo;
    SECURITY_ATTRIBUTES   saAttr;
+   char cmd_path[MAX_PATH];
 
-   const char *name = "c:\\reactos\\system32\\cmd.exe";
+   //const char *name = "c:\\reactos\\system32\\cmd.exe";
```
Comment (SergeGautherie) [inline base/services/telnetd/telnetd.c]: I assume this line is meant to be fully removed.

Comment (katahiromz) [reply base/services/telnetd/telnetd.c]: OK.

File: base/services/nfsd/idmap.c [hunk 2]
```diff
@@ -369,10 +366,13 @@ static int config_init(
         goto out;
     }
 
+    GetSystemDirectoryA(config_path, ARRAYSIZE(config_path));
+    strcat(config_path, "\\drivers\\etc\\ms-nfs41-idmap.conf");
```
Comment (HeisSpiter) [inline base/services/nfsd/idmap.c]: Check GetSystemDirectory return and use string safe functions please

File: base/services/nfsd/nfs41_daemon.c [hunk 3]
```diff
@@ -163,10 +165,23 @@ typedef struct _nfsd_args {
 static bool_t check_for_files()
 {
     FILE *fd;
+#ifdef __REACTOS__
+    char config_path[MAX_PATH];
+
+    GetSystemDirectoryA(config_path, ARRAYSIZE(config_path));
+    strcat(config_path, "\\drivers\\etc\\netconfig");
```
Comment (HeisSpiter) [inline base/services/nfsd/nfs41_daemon.c]: Check GetSystemDirectory return and use string safe functions please

File: base/services/nfsd/idmap.c [hunk 4]
```diff
@@ -73,10 +73,6 @@ struct idmap_lookup {
     const void *value;
 };
 
-
-/* configuration */
-static const char CONFIG_FILENAME[] = "C:\\ReactOS\\System32\\drivers\\etc\\ms-nfs41-idmap.conf";
```
Comment (HeisSpiter) [inline base/services/nfsd/idmap.c]: You're missing #ifdefs in that file.

File: base/services/nfsd/idmap.c (hunk 2 above)
Comment (katahiromz) [reply base/services/nfsd/idmap.c]: OK.

File: base/services/nfsd/nfs41_daemon.c (hunk 3 above)
Comment (katahiromz) [reply base/services/nfsd/nfs41_daemon.c]: OK.

File: base/services/nfsd/idmap.c (hunk 4 above)
Comment (katahiromz) [reply base/services/nfsd/idmap.c]: OK.

File: base/services/telnetd/telnetd.c [hunk 5]
```diff
@@ -367,12 +367,14 @@ static void RunShell(client_t *client)
    STARTUPINFO           si;
    PROCESS_INFORMATION   piProcInfo;
    SECURITY_ATTRIBUTES   saAttr;
+   char cmd_path[MAX_PATH];
+
+   if (!GetEnvironmentVariableA("COMSPEC", cmd_path, ARRAYSIZE(cmd_path)))
+   {
+      GetSystemDirectoryA(cmd_path, ARRAYSIZE(cmd_path));
+      strcat(cmd_path, "\\cmd.exe");
```
Comment (HeisSpiter) [inline base/services/telnetd/telnetd.c]: Sorry, forgot to review the telnet part... Can you also use StringCchCatA here and check GetSystemDirectory return?

Comment (katahiromz) [reply base/services/telnetd/telnetd.c]: OK.

File: base/services/telnetd/telnetd.h [hunk 6]
```diff
@@ -9,6 +9,7 @@
 #include <wincon.h>
 #define _INC_WINDOWS
 #include <winsock2.h>
+#include <strsafe.h>
```
Comment (SergeGautherie) [inline base/services/telnetd/telnetd.h]: Would it be enough to add this include to `telnetd.c`?

File: base/services/nfsd/idmap.c [hunk 7]
```diff
@@ -369,10 +374,29 @@ static int config_init(
...
+#ifdef __REACTOS__
+    if (GetSystemDirectoryA(config_path, ARRAYSIZE(config_path)))
+    {
+        StringCchCatA(config_path, ARRAYSIZE(config_path), "\\drivers\\etc\\ms-nfs41-idmap.conf");
+    }
+    else
+    {
+        StringCchCopyA(config_path, ARRAYSIZE(config_path), "C:\\ReactOS\\system32\\drivers\\etc\\ms-nfs41-idmap.conf");
+    }
+#endif
```
Comment (SergeGautherie) [inline base/services/nfsd/idmap.c]: Alternate logic suggestion, all occurrences:
```
if (!GetSystemDirectoryA())
   StringCchCopyA(""C:\\ReactOS\\system32");
StringCchCatA("\\...");
```

File: base/services/telnetd/telnetd.c [hunk 8]
```diff
@@ -367,12 +367,20 @@ static void RunShell(client_t *client)
...
+      {
+         StringCchCatA(cmd_path, ARRAYSIZE(cmd_path), "\\cmd.exe");
+      }
+      else
+      {
+         StringCchCopyA(cmd_path, ARRAYSIZE(cmd_path), "C:\\ReactOS\\system32\\cmd.exe");
+      }
+   }
 
-   const char *name = "c:\\reactos\\system32\\cmd.exe";
```
Comment (SergeGautherie) [inline base/services/telnetd/telnetd.c]: Should ROS have a global `#define DEFAULT_SYSTEM32_PATH "C:\\ReactOS\\System32"` (or static var, if that would apply), to use where needed? Easier to maintain/read/search/...

Comment (katahiromz) [reply base/services/telnetd/telnetd.c]: Which system header will contain it?

**Pull request merged**